cfnstack/provisioner.go

```diff
@@ -71,25 +75,25 @@ func (c *Provisioner) UploadTemplate(s3Svc S3ObjectPutterService, s3URI string,
 	return templateURL, nil
 }
 
-func (c *Provisioner) uploadTemplateIfNecessary(s3Svc S3ObjectPutterService, stackBody string, s3URI string) (*string, error) {
-	if len(stackBody) > CFN_TEMPLATE_SIZE_LIMIT {
-		if s3URI == "" {
-			return nil, fmt.Errorf("stack template's size(=%d) exceeds the 51200 bytes limit of cloudformation. `--s3-uri s3://<bucket>/path/to/dir` must be specified to upload it to S3 beforehand", len(stackBody))
-		}
-
-		templateURL, err := c.UploadTemplate(s3Svc, s3URI, stackBody)
-		if err != nil {
+func (c *Provisioner) uploadTemplates(s3Svc S3ObjectPutterService, uploads map[string]string) (*string, error) {
```

Comment:
@redbaron It turned out to me that you've used `uploadTemplates` to upload not only a template but also files referenced from it(e.g. `cloud-config-worker` and `cloud-config-controller`).
If so, I'd rather like to change its signature to something like `uploadStackAssets(s3Svc S3ObjectPutterService, stackTemplate string, cloudConfigs: map[string]string)` for now and extend/introduce a type like `StackAssets` in the future.
Then, `if filename == "stack.json" {` can be removed i.e. hence less complexity.